### PR TITLE
add tests to Vector

### LINKED ISSUE #3088: Test vectors
* Test vector service :
  * Test similarity search with different params to check if the correct return Done
  * Test similarity edge cases (long chunks) … Done
* Test **CustomSupabaseVectorStore**

> Ideally search should have a benchmark using [pytest-benchmark](https://pytest-benchmark.readthedocs.io/en/stable/) (if we have the time) 💭

### PR DESCRIPTION
**Pull request by chloedia**
closes #3088

### CODE REVIEW

File: backend/api/quivr_api/vector/tests/test_vectors.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+    session.close()
+    transaction.rollback()
+    connection.close()
+
+
+@pytest.fixture
+def test_data(session: Session) -> TestData:
+    embedding = get_embedding_client()
+    vectors = embedding.embed_documents(
```
Comment (AmineDiro) [inline backend/api/quivr_api/vector/tests/test_vectors.py:42]: This calls to openai and which should happen in unit test. use random vectors

File: backend/api/quivr_api/vector/tests/test_vectors.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    # @Amine do not get the typing() error
+
+    nested.rollback()  # Roll back the savepoint after the test
+
+
+def test_get_vectors_by_knowledge_id(session: Session, test_data: TestData):
+    vectors, knowledge, _ = test_data
+    assert knowledge.id
+
+    query = select(Vector).where(Vector.knowledge_id == knowledge.id)
```
Comment (AmineDiro) [inline backend/api/quivr_api/vector/tests/test_vectors.py:99]: should have a higher level construct in repo for this ?

File: backend/api/quivr_api/vector/tests/test_vectors.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        knowledge_id=knowledge_1.id,
+    )
+
+    session.add(vector_1)
+    session.add(vector_2)
+
+    session.flush()
+
+    yield ([vector_1, vector_2], knowledge_1, brain_1)
+    # @Amine do not get the typing() error
```
Comment (AmineDiro) [inline backend/api/quivr_api/vector/tests/test_vectors.py:90]: Are you assigning this to me in your PR ?

**Review by AmineDiro — COMMENTED** [review]
Some additional tests required. We should test  :
- Create_vectors in vector_service 
- Similarity serach of service. 

You could should to implement for service /and repo but we have additional logic in the vector service that should be tested. The search is critical for RAG so we need to have a higher coverage.

**Comment by chloedia** [comment]
> Some additional tests required. We should test :
> 
> * Create_vectors in vector_service


Create vector un the service is using open ai embeddings, and you said test shouldn't call the open ai api !
